src/cargo/ops/cargo_rustc/mod.rs

```diff
@@ -420,6 +423,21 @@ fn rustc(cx: &mut Context, unit: &Unit, exec: Arc<Executor>) -> CargoResult<Work
         }
         Ok(())
     }
+
+    // Add all custom environment variables present in `state` (after they've
+    // been put there by one of the `build_scripts`) to the command provided.
+    fn add_custom_env(rustc: &mut ProcessBuilder,
+                      build_state: &BuildMap,
+                      _: &BuildScripts,
+                      current_id: &PackageId) -> CargoResult<()> {
+        let key = (current_id.clone(), Kind::Host);
```

Comment:
Ah yes here you'll want to pass in a `kind` instead of just picking `Kind::Host`, which I believe will break cross-compiled builds. Typically most rules have `unit.kind` which can be ferried to this locaion.